homework_21/homework_21.txt

```diff
@@ -0,0 +1,47 @@
+Використовуючи базу даних з додаткових матеріалів лекції:
+
+1. Написати sql запит, який вибере усі записи із таблиці users старше 30 років.
+2. Написати sql запит, який виведе кількість записів в табліці users, що старше 30 років.
+3. Написати sql запит, який виведе інформацію про вік (кількість років) та
+кількість користувачів, які відповідають цьому віку. Результатом виконання такого запиту
+має бути таблиця:
+    age  | users
+    32    | 1
+    52    | 2
+    120  | 2
+    1142 | 1
+
+4. Написати sql запит, який буде робити те саме, що і в завданні 3, але виводити дані
+відсортовані по кількості користувачів у спадаючому порядку та по віку у
+зростаючому порядку. Результатом має бути таблиця:
+    age  | users
+    52    | 2
+    120  | 2
+    32    | 1
+    1142 | 1
+
+5. (необов'язкове виконання) Модифікувати попередній запит таким чином,
+щоб з отриманого результату вибрати тільки ті записи, де значення users більше 1.
+Результатом має бути таблиця:
+    age  | users
+    52    | 2
+    120  | 2
+
+    1
+SELECT * FROM users WHERE age > 30;
+
+    2
+SELECT COUNT(*) FROM users WHERE age > 30;
+
+    3
+SELECT age, COUNT(age) FROM users WHERE age > 30 GROUP BY age;
+
+    4
+SELECT age, COUNT(age) FROM users WHERE age > 30 GROUP BY age ORDER BY COUNT(age) desc , age ;
```

Comment:
You can use aliases when you need to reuse a result of a function:
`... COUNT(age) as age_count ... ORDER BY age_count ...`

Reply:
fixed